cmd/kured/main.go

```diff
@@ -652,11 +670,9 @@ func rebootAsRequired(nodeID string, rebootCommand []string, sentinelCommand []s
 			// And finally (3) if it has that annotation, to delete it.
 			// This indicates to other node tools running on the cluster that this node may be a candidate for maintenance
 			if annotateNodes && !rebootRequired(sentinelCommand) {
-				if _, ok := node.Annotations[KuredRebootInProgressAnnotation]; ok {
-					err := deleteNodeAnnotation(client, nodeID, KuredRebootInProgressAnnotation)
-					if err != nil {
-						continue
-					}
+				err := deleteNodeAnnotation(client, nodeID, KuredRebootInProgressAnnotation)
```

Comment:
Why the if was removed? It would avoid a call to the apiserver.

Reply:
Readded the check.